pkg/test/framework/components/ingress/kube.go

```diff
@@ -120,28 +157,92 @@ func (c *kubeComponent) Address() string {
 	return c.address
 }
 
-func (c *kubeComponent) Call(path string) (CallResponse, error) {
+// createClient creates a client which sends HTTP requests or HTTPS requests, depending on
+// ingress type. If host is not empty, the client will resolve domain name and verify server
+// cert using the host name.
+func (c *kubeComponent) createClient(host string) (*http.Client, error) {
 	client := &http.Client{
 		Timeout: 1 * time.Minute,
 	}
+	if c.gatewayType != PlainText {
+		scopes.Framework.Debug("Prepare root cert for client")
+		roots := x509.NewCertPool()
+		ok := roots.AppendCertsFromPEM([]byte(c.caCert))
+		if !ok {
+			return nil, errors.New("failed to parse root certificate")
+		}
+		tlsConfig := &tls.Config{
+			RootCAs:    roots,
+			ServerName: host,
+		}
+		if c.gatewayType == Mtls {
+			cer, err := tls.X509KeyPair([]byte(c.tlsCert), []byte(c.tlsKey))
+			if err != nil {
+				return nil, errors.New("failed to parse private key and server cert")
+			}
+			tlsConfig.Certificates = []tls.Certificate{cer}
+		}
+		tr := &http.Transport{
+			TLSClientConfig: tlsConfig,
+			DialTLS: func(netw, addr string) (net.Conn, error) {
+				if addr == host+":443" {
+					addr = c.address + ":443"
+				}
+				tc, err := tls.Dial(netw, addr, tlsConfig)
+				if err != nil {
+					scopes.Framework.Errorf("TLS dial fail: %v", err)
+					return nil, err
+				}
+				if err := tc.Handshake(); err != nil {
+					scopes.Framework.Errorf("SSL handshake fail: %v", err)
+					return nil, err
+				}
+				return tc, nil
+			}}
+		client.Transport = tr
+	}
+	return client, nil
+}
 
+// createRequest returns a request for client to send, or nil and error if request is failed to generate.
+func (c *kubeComponent) createRequest(path, host string) (*http.Request, error) {
 	if !strings.HasPrefix(path, "/") {
 		path = "/" + path
 	}
 	url := c.address + path
-
-	scopes.Framework.Debugf("Sending call to ingress at: %s", url)
+	if c.gatewayType != PlainText {
+		url = "https://" + host + ":443" + path
+	}
 
 	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
+		return nil, err
+	}
+	if c.gatewayType != PlainText && host != "" {
+		req.Host = host
+	}
+
+	scopes.Framework.Debugf("Created a request to send %v", req)
+	return req, nil
+}
+
+func (c *kubeComponent) Call(path, host string) (CallResponse, error) {
+	client, err := c.createClient(host)
+	if err != nil {
+		scopes.Framework.Errorf("failed to create test client, error %v", err)
+		return CallResponse{}, err
+	}
+	req, err := c.createRequest(path, host)
+	if err != nil {
+		scopes.Framework.Errorf("failed to create request, error %v", err)
 		return CallResponse{}, err
 	}
 
 	resp, err := client.Do(req)
```

Comment:
Do we need retries?

Reply:
Retry is done in VisitProductPage() at sds_ingress/util/util.go.
I want to avoid too much change to the Call() method, as it is already used in other test.